### PR TITLE
Document System.IO.MemoryStream

### PR DESCRIPTION
**Pull request by carlossanlop**
Jeremy and I are the area owners.

### CODE REVIEW

File: xml/System.IO/MemoryStream.xml [hunk 1]
```diff
@@ -544,14 +544,26 @@
...
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+Refer to the <xref:System.IO.Stream.BeginRead> remarks for additional usage information for this method.
+
+          ]]></format>
+        </remarks>
+        <exception cref="T:System.IO.IOException">Attempted an asynchronous read past the end of the memory stream, or a disk error occurs.</exception>
```
Comment (gewarren) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
        <exception cref="T:System.IO.IOException">Attempted an asynchronous read past the end of the memory stream, or a disk error occurred.</exception>
```

File: xml/System.IO/MemoryStream.xml [hunk 2]
```diff
@@ -588,14 +600,26 @@
...
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+Refer to the <xref:System.IO.Stream.BeginWrite> remarks for additional usage information for this method.
+
+          ]]></format>
+        </remarks>
+        <exception cref="T:System.IO.IOException">Attempted an asynchronous write past the end of the memory stream, or a disk error occurs.</exception>
```
Comment (gewarren) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
        <exception cref="T:System.IO.IOException">Attempted an asynchronous write past the end of the memory stream, or a disk error occurred.</exception>
```

File: xml/System.IO/MemoryStream.xml [hunk 3]
```diff
@@ -544,14 +544,26 @@
...
-        <param name="state">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="buffer">The buffer to read the data into.</param>
+        <param name="offset">The byte offset in <paramref name="buffer" /> at which to begin writing data read from the memory stream.</param>
+        <param name="count">The maximum number of bytes to read.</param>
+        <param name="callback">An optional asynchronous callback, to be called when the read is complete.</param>
+        <param name="state">A user-provided object that distinguishes this particular asynchronous read request from other requests.</param>
+        <summary>Begins an asynchronous read operation. (Consider using <see cref="M:System.IO.MemoryStream.ReadAsync(System.Byte[],System.Int32,System.Int32)" /> instead.)</summary>
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]: The build is throwing warnings. Should this link to the following API instead?
```suggestion
        <summary>Begins an asynchronous read operation. (Consider using <see cref="M:System.IO.MemoryStream.ReadAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)" /> instead.)</summary>
```

File: xml/System.IO/MemoryStream.xml [hunk 4]
```diff
@@ -544,14 +544,26 @@
...
+        <returns>An <see cref="T:System.IAsyncResult" /> that represents the asynchronous read, which could still be pending.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+Refer to the <xref:System.IO.Stream.BeginRead> remarks for additional usage information for this method.
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
Refer to the <xref:System.IO.Stream.BeginRead%2A> remarks for additional usage information for this method.
```

File: xml/System.IO/MemoryStream.xml [hunk 5]
```diff
@@ -588,14 +600,26 @@
...
+        <param name="callback">An optional asynchronous callback, to be called when the write is complete.</param>
+        <param name="state">A user-provided object that distinguishes this particular asynchronous write request from other requests.</param>
+        <summary>Begins an asynchronous write operation. (Consider using <see cref="M:System.IO.MemoryStream.WriteAsync(System.Byte[],System.Int32,System.Int32)" /> instead.)</summary>
+        <returns>An <see langword="IAsyncResult" /> that represents the asynchronous write, which could still be pending.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+Refer to the <xref:System.IO.Stream.BeginWrite> remarks for additional usage information for this method.
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
Refer to the <xref:System.IO.Stream.BeginWrite%2A> remarks for additional usage information for this method.
```

File: xml/System.IO/MemoryStream.xml [hunk 6]
```diff
@@ -588,14 +600,26 @@
...
-        <param name="state">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="buffer">The buffer to write data from.</param>
+        <param name="offset">The byte offset in <paramref name="buffer" /> from which to begin writing.</param>
+        <param name="count">The maximum number of bytes to write.</param>
+        <param name="callback">An optional asynchronous callback, to be called when the write is complete.</param>
+        <param name="state">A user-provided object that distinguishes this particular asynchronous write request from other requests.</param>
+        <summary>Begins an asynchronous write operation. (Consider using <see cref="M:System.IO.MemoryStream.WriteAsync(System.Byte[],System.Int32,System.Int32)" /> instead.)</summary>
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]: this API instead?

```suggestion
        <summary>Begins an asynchronous write operation. (Consider using <see cref="M:System.IO.MemoryStream.WriteAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)" /> instead.)</summary>
```

File: xml/System.IO/MemoryStream.xml [hunk 7]
```diff
@@ -911,10 +935,16 @@ End Try
...
-        <summary>To be added.</summary>
+        <param name="destination">The stream to which the contents of the current memory stream will be copied.</param>
+        <param name="bufferSize">The size of the buffer. This value must be greater than zero. The default size is 81920.</param>
+        <summary>Reads the bytes from the current memory stream and writes them to another stream, using a specified buffer size.</summary>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException"><paramref name="destination" /> is <see langword="null" />.</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException"><paramref name="bufferSize" /> is not a positive number.</exception>
+        <exception cref="T:System.ObjectDisposedException">Either the underlying memory stream or the <paramref name="destination" /> stream is closed.</exception>
+        <exception cref="T:System.NotSupportedException">The underlying memory stream is unreadable.
+-or-
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion

-or-

```

File: xml/System.IO/MemoryStream.xml [hunk 8]
```diff
@@ -1055,10 +1085,25 @@ End Try
...
+
+## Remarks
+
+Refer to the <xref:System.IO.Stream.EndRead> remarks for additional usage information for this method.
+
+          ]]></format>
+        </remarks>
+        <exception cref="T:System.ArgumentNullException"><paramref name="asyncResult" /> is <see langword="null" />.</exception>
+        <exception cref="T:System.ArgumentException">A handle to the pending read operation is not available.
+-or-
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion

-or-
```

File: xml/System.IO/MemoryStream.xml [hunk 9]
```diff
@@ -1055,10 +1085,25 @@ End Try
         <Parameter Name="asyncResult" Type="System.IAsyncResult" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1" />
       </Parameters>
       <Docs>
-        <param name="asyncResult">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="asyncResult">The reference to the pending asynchronous request to finish.</param>
+        <summary>Waits for the pending asynchronous read to complete. (Consider using <see cref="M:System.IO.MemoryStream.ReadAsync(System.Byte[],System.Int32,System.Int32)" /> instead.)</summary>
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
        <summary>Waits for the pending asynchronous read to complete. (Consider using <see cref="M:System.IO.MemoryStream.ReadAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)" /> instead.)</summary>
```

File: xml/System.IO/MemoryStream.xml [hunk 10]
```diff
@@ -1055,10 +1085,25 @@ End Try
...
+        <returns>The number of bytes read from the memory stream, between zero (0) and the number of bytes you requested. Returns zero (0) only at the end of the memory stream, otherwise, it should block until at least one byte is available.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+Refer to the <xref:System.IO.Stream.EndRead> remarks for additional usage information for this method.
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
Refer to the <xref:System.IO.Stream.EndRead%2A> remarks for additional usage information for this method.
```

File: xml/System.IO/MemoryStream.xml [hunk 11]
```diff
@@ -1091,9 +1136,25 @@ End Try
         <Parameter Name="asyncResult" Type="System.IAsyncResult" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1" />
       </Parameters>
       <Docs>
-        <param name="asyncResult">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="asyncResult">A reference to the outstanding asynchronous I/O request.</param>
+        <summary>Ends an asynchronous write operation. (Consider using <see cref="M:System.IO.MemoryStream.WriteAsync(System.Byte[],System.Int32,System.Int32)" /> instead.)</summary>
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
        <summary>Ends an asynchronous write operation. (Consider using <see cref="M:System.IO.MemoryStream.WriteAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)" /> instead.)</summary>
```

File: xml/System.IO/MemoryStream.xml [hunk 12]
```diff
@@ -1091,9 +1136,25 @@ End Try
...
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+Refer to the <xref:System.IO.Stream.EndWrite> remarks for additional usage information for this method.
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
Refer to the <xref:System.IO.Stream.EndWrite%2A> remarks for additional usage information for this method.
```

**Pull request merged**